### ISSUE TITLE
iPad: improve browsing experience when no real keyboard is available

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode#125306]
When no "real" keyboard is attached to the iPad (and therefore keyboard shortcuts aren't available), it is a pain to activate some frequently used actions like the "Command palette". Delving into the (tiny) "View" submenu in the "Hamburger" menubar and then finding and hitting the "Command Palette" action is hit-and-miss operation...

It would great if the menubar could use an iPad (or in general: "touch") specific UI:
e.g. a 3-finger swipe would open a large overlay where all frequently used menu actions are available with a single tap.

just dreaming...

**Comment by isidorn** [comment]
Great idea. A good start would be for a 3-finger swipe to just open the quick pick.
I can look into if it is possible to detect the 3 finger gesture from the browser.

Another idea is to simply put a quick pick action in the activity bar (but I think there is a separate issue for this).

fyi @TylerLeonhardt for quick pick

**Comment by weinand** [comment]
Yes, having a gesture to open the "command palette" is a first step.

But having a touch-based replacement for the tiny menubar with all the tiny submenus is another big issue...
This UI is just not "touch"-ready.

**Comment by TylerLeonhardt** [comment]
> A good start would be for a 3-finger swipe to just open the quick pick.

I wonder if the "omnibar in the title bar of vscode" is the way to go here. It's something @misolori mocked up that might be good for this

**Comment by isidorn** [comment]
@weinand for the menus I already created this issue https://github.com/microsoft/vscode/issues/121202
I am not sure how far we can go with making the menu larger. To be honest trying it out on the large iPad pro, the menus where not that bad. I was not convinced we need a brand new experience. Though it might be needed.

@TylerLeonhardt omnibar would obviously be best. But a good first step is to just show the quick pick. And once we have the omnibar we can start showing it instead.

**Comment by isidorn** [comment]
Here's a poitner for how we could try to detect gesture and based on them open Quick Pick https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html#//apple_ref/doc/uid/TP40006511-SW23

Not sure if it would work
Also fyi @bpasero for ideas that a three finger gesture would open quick pick on iPad

**Comment by isidorn** [comment]
Nothing for this milestone. Moving out to backlog.
@TylerLeonhardt also assigning to you since you own the Quick Pick and maybe this could be interesting? Let me know what you think.
